packages/shared/src/types/PostCollectionConfiguration.ts

```diff
@@ -0,0 +1,166 @@
+import * as db from '../db';
+import * as ChannelAction from './ChannelActions';
+
+/**
+ * How do we want to lay out a collection of posts, at a high level?
+ */
+export type PostCollectionLayoutType =
+  | 'comfy-list-top-to-bottom' // think: notebook
+  | 'compact-list-bottom-to-top' // think: chat
+  | 'grid'; // think: gallery
+
+// Why overload this function instead of just doing a union?
+// If the caller has a non-nullable `channel`, they can then get a
+// non-nullable return value - nice, right?
+export function postCollectionLayoutTypeFromChannel(
```

Comment:
these function names are pretty long. I think they'd survive removing `postCollection` from the front and still be descriptive enough.

Reply:
a post has a layout as well; but I'll concede, given that the functions are in `PostCollectionConfiguration.ts`. (if the team doesn't like long identifiers, maybe we should adopt https://eslint.org/docs/latest/rules/id-length )

Reply:
[ae9baaa](https://github.com/tloncorp/tlon-apps/pull/3993/commits/ae9baaa4a8d7ed1c5a4424170b71c3ae1943d836)